File: problems/012_XXX.java
public class Problemn {
	public static void main(String[] args) {
        Integer triangleNumber = 0;
        Integer counter = 0;
        Integer ndivisors = 0;
        while (ndivisors < 50) {
            counter++;
            triangleNumber += counter;
            ndivisors = count_divisors(triangleNumber);
    		//System.out.println(triangleNumber);
        }
		System.out.println(triangleNumber);
	}

    public static Integer count_divisors(Integer n){
        Integer ndivisors = 0;
        for (Integer i = 1; i <= n; i++){
            if (n % i == 0) {
                ndivisors++;
            }
        }
        return ndivisors;
    }
}

File: problems/012_XXX.c
#include <stdio.h>
#include <math.h>

unsigned int count_divisors(unsigned long n){
    unsigned int n_divisors = 0;
    for (unsigned long i = 1; i <= n; i++) {
        if (n % i == 0){
            int k = 0;
            n_divisors++;
        }
    }
    return n_divisors;
}

void main(){
    unsigned long triangle_number = 0;
    unsigned long counter = 0;
    unsigned int n_divisors = 0;
    while (n_divisors < 500) {
        counter++;
        triangle_number += counter;
        n_divisors = count_divisors(triangle_number);
        //printf("%lu\n", triangle_number);
    }
    printf("%lu\n", triangle_number);
}